### DISCUSSION TITLE
different quality between reference and recognition

### DISCUSSION THREAD

**Discussion by derjape** [jomjol/AI-on-the-edge-device#1826] [Q&A]
Hi,

I'm a little bit confused about the fact that the image I created during "set reference" has good quality and the settings according to brightness contrast etc are used correct. But the Image that is taken under "Overview" and "Recognition" is from poor quality (I think less resolution) and not at the correct brightness .

Is this a problem or bug? or is it a complete correct behavior??
![Create_reference](https://user-images.githubusercontent.com/83368666/212417596-efdc980b-414c-4bbc-869b-f6fff32c6163.png)
![recognition](https://user-images.githubusercontent.com/83368666/212417605-8e3d74c9-7a58-4c56-9844-ac8136967d54.png)

**Comment by caco3** [comment]
I think it is for performance reasons.

I thought I read something about this before but could not find it on a quick search.

**Reply by derjape** [reply]
@caco3 just for clarification:
do you mean, the "recognition image" is reduced in quality because the performance of the ESP32 and the AI needs this (it is a normal behavior)
or do you mean, it is reduced quality because my system is running (for whatever reason) with limited performance (due to some reason the performance is reduced)?

**Reply by caco3** [reply]
If remember correctly, the Reference Image is taken with a higher resolution than the later image during a normal round.
And that is normal behaviour for all of us.

**Comment by Slider0007** [comment]
I also could be that saturation and contrast is not fully supported right now, but up to now I haven't checked  in detail (To keep track I opened a feature request #1851)
![image](https://user-images.githubusercontent.com/115730895/212560719-17c5a1e1-e672-45b5-809f-7ebd76a8b13c.png)

So it could be that during creating reference saturation and contrast setting is used but during the rounds it's ingoring it. I assume @jomjol could maybe have more details...

@derjape: You could test with saturation and contrast setting to 0.

**Reply by derjape** [reply]
@Slider0007 thanks for that hint.
actually I'm fighting with external LED illumination.  So I will check saturation, brightness and contrast =ZERO within the next 2 or 3 days and report my observations.

**Comment by jomjol** [comment]
There should be no difference for the image taking during the Setup and the Running later on.
The parameters for the cameras are not fully supported in both situation.

There is a difference with respect to rotate the image. In the Take Image configuration it is done within the browser with anti alising. Later on it is done on the ESP32, giving some minor artifacts.
But that is not an explanation for the brightness changes. Up to know I have no clear hint, why there is this difference

**Comment by bucklevision** [comment]
Alignment image:
![Screenshot_20230120_150213_com android chrome](https://user-images.githubusercontent.com/29090176/213732046-db187c48-a165-450b-b5e2-bfd206f42b44.jpg)
Actual image, with aliasing and some level of warp:
![Screenshot_20230120_150229_com android chrome](https://user-images.githubusercontent.com/29090176/213732061-ba7a2c50-84a1-400a-990e-7a4c470067b2.jpg)
